From 1e63fa7c70f44c7b61e881058b472f493fc8ef71 Mon Sep 17 00:00:00 2001
From: Weili Qian <qianweili@huawei.com>
Date: Wed, 9 Mar 2022 10:15:35 +0800
Subject: [PATCH] crypto: hisilicon/qm - modify device status check parameter
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 1e63fa7c70f44c7b61e881058b472f493fc8ef71
Modified-by-SEL: No


driver inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4W3OQ

----------------------------------------------------------------------

If the device master ooo is blocked, there is
no need to empty the queue. Only the PF can obtain the
status of the device. If the VF runs on the host,
the device status can be obtained by PF.

Signed-off-by: Weili Qian <qianweili@huawei.com>
Signed-off-by: Yang Shen <shenyang39@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/crypto/hisilicon/qm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/hisilicon/qm.c b/drivers/crypto/hisilicon/qm.c
index 95d61fd62074..89e71a3878a6 100644
--- a/drivers/crypto/hisilicon/qm.c
+++ b/drivers/crypto/hisilicon/qm.c
@@ -2858,6 +2858,7 @@ static int qm_drain_qp(struct hisi_qp *qp)
 {
 	size_t size = sizeof(struct qm_sqc) + sizeof(struct qm_cqc);
 	struct hisi_qm *qm = qp->qm;
+	struct hisi_qm *pf_qm = pci_get_drvdata(pci_physfn(qm->pdev));
 	struct device *dev = &qm->pdev->dev;
 	struct qm_sqc *sqc;
 	struct qm_cqc *cqc;
@@ -2866,7 +2867,7 @@ static int qm_drain_qp(struct hisi_qp *qp)
 	void *addr;
 
 	/* No need to judge if master OOO is blocked. */
-	if (qm_check_dev_error(qm))
+	if (qm_check_dev_error(pf_qm))
 		return 0;
 
 	/* Kunpeng930 supports drain qp by device */
-- 
2.26.2

